src/Compilers/CSharp/Portable/SymbolDisplay/SymbolDisplayVisitor.Types.cs

```diff
@@ -547,36 +547,52 @@ private void AddTypeKind(INamedTypeSymbol symbol)
                 }
                 else
                 {
-                    var kindKeyword = GetKindKeyword(symbol.TypeKind);
-                    if (kindKeyword != SyntaxKind.None)
+                    switch (symbol.TypeKind)
                     {
-                        AddKeyword(kindKeyword);
-                        AddSpace();
+                        case TypeKind.Module:
+                        case TypeKind.Class:
+                            AddKeyword(SyntaxKind.ClassKeyword);
+                            AddSpace();
+                            break;
+
+                        case TypeKind.Enum:
+                            AddKeyword(SyntaxKind.EnumKeyword);
+                            AddSpace();
+                            break;
+
+                        case TypeKind.Delegate:
+                            AddKeyword(SyntaxKind.DelegateKeyword);
+                            AddSpace();
+                            break;
+
+                        case TypeKind.Interface:
+                            AddKeyword(SyntaxKind.InterfaceKeyword);
+                            AddSpace();
+                            break;
+
+                        case TypeKind.Struct:
+                            var structSymbol = (NamedTypeSymbol)symbol;
```

Comment:
Could the symbol be a `VisualBasic.Symbols.NamedTypeSymbol`?

Reply:
@cston This is C# compiler project. Not sure how can symbols leak?

Reply:
`SymbolDisplay` is a public API and can be passed any `INamedTypeSymbol` implementation. There are several tests in `SymbolDisplayTests` that mix C# and VB.

Reply:
I see. Will add a check for that.